frontend/src/components/Popup.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useContext, useEffect, useRef } from 'react';
+import { MenuContext } from '../context/MenuContext';
+import Leaderboard from './popups/Leaderboard';
+import Settings from './popups/Settings';
+import Info from './popups/Info';
+import Signup_Login from './popups/Signup_Login';
+import './Component.css';
+
+function Popup() {
```

Comment:
Should we name this file PopupWrapper instead?

Reply:
We could, I think that either name works for this component.